histogram/benches/histogram.rs

```diff
@@ -0,0 +1,99 @@
+use core::time::Duration;
+use criterion::{criterion_group, criterion_main, Criterion, Throughput};
+
+fn histogram(c: &mut Criterion) {
+    let mut histogram = histogram::Histogram::new(0, 7, 64).unwrap();
+
+    let mut group = c.benchmark_group("histogram");
+    group.throughput(Throughput::Elements(1));
+    group.bench_function("increment/1", |b| b.iter(|| histogram.increment(1)));
+
+    histogram.clear();
+    group.bench_function("increment/max", |b| {
+        b.iter(|| histogram.increment(u64::MAX))
+    });
+
+    group.finish();
+}
+
+fn histogram_atomic(c: &mut Criterion) {
+    let histogram = histogram::atomic::Histogram::new(0, 7, 64).unwrap();
+
+    let mut group = c.benchmark_group("histogram::atomic");
+    group.throughput(Throughput::Elements(1));
+    group.bench_function("increment/1", |b| b.iter(|| histogram.increment(1)));
+
+    histogram.clear();
+    group.bench_function("increment/max", |b| {
+        b.iter(|| histogram.increment(u64::MAX))
+    });
+
+    group.finish();
+}
+
+fn sliding_window(c: &mut Criterion) {
+    // millisecond resolution
+
+    let mut histogram =
+        histogram::sliding_window::Histogram::new(0, 7, 64, Duration::from_millis(1), 100).unwrap();
+
+    let mut group = c.benchmark_group("histogram::sliding_window/milliseconds");
```

Comment:
So... we can't parameterize it exactly because there is not (and cannot be?) a unifying trait for all histograms. Notably, the non-atomic versions take a mutable reference for operations that change state (eg: increment) while the atomic ones take an immutable reference (since they use interior mutability via atomics). 

I agree the duplication is annoying. So, I have written a macro and we can just pretend these differences don't exist.